predicates/forwarded/forwarded.go

```diff
@@ -0,0 +1,150 @@
+/*
+Package forwarded implements a set of custom predicate to match routes
+based on the standardized Forwarded header.
+
+https://datatracker.ietf.org/doc/html/rfc7239
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded
+
+Examples:
+
+    // only match requests to "example.com"
+    example1: ForwardedHost("example.com") -> "http://example.org";
+
+    // only match requests to http
+    example2: ForwardedProtocol("http") -> "http://example.org";
+
+    // only match requests to https
+    example3: ForwardedProtocol("https") -> "http://example.org";
+*/
+package forwarded
+
+import (
+	"github.com/zalando/skipper/predicates"
+	"github.com/zalando/skipper/routing"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	NameHost  = "ForwardedHost"
+	NameProto = "ForwardedProtocol"
+)
+
+type hostPredicateSpec struct{}
+
+type protoPredicateSpec struct{}
+
+type hostPredicate struct {
+	host *regexp.Regexp
+}
+
+func (p *hostPredicateSpec) Create(args []interface{}) (routing.Predicate, error) {
+	if len(args) != 1 {
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	value, ok := args[0].(string)
+	if !ok {
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	if value == "" {
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	re, err := regexp.Compile(value)
+	if err != nil {
+		return nil, err
+	}
+
+	return hostPredicate{host: re}, err
+}
+
+type protoPredicate struct {
+	proto string
+}
+
+func (p *protoPredicateSpec) Create(args []interface{}) (routing.Predicate, error) {
+
+	if len(args) != 1 {
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	value, ok := args[0].(string)
+	if !ok {
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	value = strings.ToLower(value)
+
+	switch value {
+	case "http", "https":
+	default:
+		return nil, predicates.ErrInvalidPredicateParameters
+	}
+
+	return protoPredicate{proto: value}, nil
```

Comment:
minor: I think `ToLower` is redundant - the docs explicitly say it should be lowercase. 
If we change case people eventually would rely on it.
I would also move last return inside the switch:
```go
        switch value {
	case "http", "https":
        	return protoPredicate{proto: value}, nil
	default:
		return nil, predicates.ErrInvalidPredicateParameters
	}
```

Reply:
True and it should be handled by this:
```
func (h Header) Get(key string) string
    Get gets the first value associated with the given key. If there are no
    values associated with the key, Get returns "". It is case insensitive;
    textproto.CanonicalMIMEHeaderKey is used to canonicalize the provided key.
    To use non-canonical keys, access the map directly.
```

Reply:
Not sure `func (h Header) Get(key string) string` handles this. The casing (lowercasing in this case) was for the parameter for the predicate, not for the actual value of the header.

Reply:
Removed the lowercase, and simplified the code as suggested.